instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -57,7 +57,7 @@ def connect
                 OpenTelemetry::SemanticConventions::Trace::NET_PEER_PORT => conn_port
               }.merge!(OpenTelemetry::Common::HTTP::ClientContext.attributes)
 
-              tracer.in_span('HTTP CONNECT', attributes: attributes) do
+              tracer.in_span('HTTP SESSION', attributes: attributes) do
```

Comment:
The recommendation in the semantic conventions is:
> HTTP client spans SHOULD be using conservative, low cardinality names formed from the available parameters of an HTTP request, such as `"HTTP {METHOD_NAME}"`.

While this is a recommendation, it's reasonable for users to assume HTTP client spans named `HTTP SESSION` represent HTTP requests with a method of `SESSION`, which ... isn't a thing.

I'd prefer it we checked for `use_ssl? && proxy?` and kept `HTTP CONNECT` as the span name in that case. If we also want to trace session establishment in the non-SSL-proxy case, we should probably do that with a different span. Ideally, that'd be an internal span (I think) that is always created on this path, wrapping the `HTTP CONNECT` client span in the SSL-proxy case, but that's challenging to implement without being incredibly brittle. In any case, it likely shouldn't be a HTTP client span, since it is lower-level and doesn't actually speak HTTP.

Reply:
> While this is a recommendation, it's reasonable for users to assume HTTP client spans named HTTP SESSION represent HTTP requests with a method of SESSION, which ... isn't a thing.

💯 I mentioned to the folks on the SIG call today that this was a terrible name and I was looking for ideas on a better name. I figured it was better to push out a PR with a bad name and get some feedback from folks and see if we could come up with a better name. 

> I'd prefer it we checked for use_ssl? && proxy? and kept HTTP CONNECT as the span name in that case. If we also want to trace session establishment in the non-SSL-proxy case, we should probably do that with a different span. Ideally, that'd be an internal span (I think) that is always created on this path, wrapping the HTTP CONNECT client span in the SSL-proxy case, but that's challenging to implement without being incredibly brittle. In any case, it likely shouldn't be a HTTP client span, since it is lower-level and doesn't actually speak HTTP.

Are you are asking me to include this implementation as part of this PR? i.e. conditionally add an `HTTP CONNECT` PR as well as add an internal spans to trace session establishment? Or is renaming the existing span OK?

Reply:
added a few more branches there.